app/liquid_tags/wikipedia_tag.rb

```diff
@@ -0,0 +1,87 @@
+class WikipediaTag < LiquidTagBase
+  PARTIAL = "liquids/wikipedia".freeze
+
+  def initialize(tag_name, input, tokens)
+    super
+    @data = get_data(input.strip)
+  end
+
+  def render(_context)
+    ActionController::Base.new.render_to_string(
+      partial: PARTIAL,
+      locals: {
+        title: @data[:title],
+        extract: @data[:extract],
+        url: @data[:url]
+      },
+    )
+  end
+
+  private
+
+  def valid_url?(input)
+    (input =~ /^https?\:\/\/([a-z-]+)\.wikipedia.org\/wiki\/([\S]+)$/)&.zero?
+  end
+
+  def get_data(input)
+    sanitized_url = ActionController::Base.helpers.strip_tags(input).strip
+    raise StandardError, "Invalid Wikipedia share URL" unless valid_url?(sanitized_url)
+
+    uri = URI.parse(sanitized_url)
+    lang = uri.host.split(".").first
+    title = uri.path.split("/").last
+    anchor = uri.fragment
+
+    if anchor
+      response = HTTParty.get("https://#{lang}.wikipedia.org/api/rest_v1/page/mobile-sections/#{title}", format: :json)
+      handle_response_error(response, sanitized_url)
+
+      text, section_title = get_section_contents(response, anchor, sanitized_url)
+      title = "#{response['lead']['normalizedtitle']} - #{section_title}"
+      extract = text_clean_up(text)
+    else
+      response = HTTParty.get("https://#{lang}.wikipedia.org/api/rest_v1/page/summary/#{title}", format: :json)
+      handle_response_error(response, sanitized_url)
+
+      title = response["title"]
+      extract = response["extract"]
+    end
+
+    {
+      title: title,
+      url: sanitized_url,
+      extract: extract
+    }
+  end
+
+  def handle_response_error(response, input)
+    raise StandardError, "Couldn't find the Wikipedia article {% #{tag_name} #{input} %}: #{response['detail']}" if response.code != 200
+  end
+
+  def get_section_contents(response, anchor, input)
+    text = title = ""
+    response["remaining"]["sections"].each do |section|
+      if section["anchor"] == CGI.unescape(anchor)
+        text = section["text"]
+        title = section["line"]
+      end
+    end
+
+    raise StandardError, "Couldn't find the section of the Wikipedia article {% #{tag_name} #{input} %}" if title.blank?
```

Comment:
Same, please don't use modifiers on long lines, they are easy to miss when reading the code.